### PR TITLE
Add JavaDocs for base and common modules

### PR DESCRIPTION
**Pull request by dliu-workos**

### CODE REVIEW

File: workos/src/main/kotlin/com/workos/WorkOS.kt [hunk 1]
```diff
@@ -23,66 +22,105 @@ import java.net.http.HttpRequest
...
 
+/**
+ * Global configuration class for interacting with the WorkOS API.
+ *
+ * @param apiKey The API Key used for authenticating requests.
+ */
 class WorkOS(
   val apiKey: String
 ) {
-  init {
```
Comment (dliu-workos) [inline workos/src/main/kotlin/com/workos/WorkOS.kt:29]: The deletes are just re-ordering code

File: workos/src/main/kotlin/com/workos/WorkOS.kt [hunk 2]
```diff
@@ -164,12 +211,7 @@ class WorkOS(
       }
       else -> {
         val responseData = mapper.readValue(response.body(), GenericErrorResponse::class.java)
-
-        if (responseData.error != null || responseData.error_description != null) {
-          throw OauthException(responseData.message, status, requestId, responseData.error, responseData.error_description)
```
Comment (dliu-workos) [inline workos/src/main/kotlin/com/workos/WorkOS.kt:169]: Deleted this -- we do not need it now that we are enforcing an API key in the constructor.

File: workos/src/main/kotlin/com/workos/WorkOS.kt [hunk 3]
```diff
@@ -23,66 +22,105 @@ import java.net.http.HttpRequest
...
-  }
 
+  /**
+   * Host to send requests to.
+   */
+  @JvmField
   var apiHostname = "api.workos.com"
 
+  /**
+   * Whether or not to use the HTTPs for requests.
```
Comment (mthadley) [inline workos/src/main/kotlin/com/workos/WorkOS.kt:41]: Maybe this instead?

```suggestion
   * Whether or not to use HTTPS for requests.
```

File: workos/src/main/kotlin/com/workos/common/http/PaginationParams.kt [hunk 4]
```diff
@@ -1,5 +1,12 @@
 package com.workos.common.http
 
+/**
+ * Parameters for performing paginated requested on lists of resources.
```
Comment (mthadley) [inline workos/src/main/kotlin/com/workos/common/http/PaginationParams.kt:4]:
```suggestion
 * Parameters for performing paginated requests on lists of resources.
```

File: workos/src/main/kotlin/com/workos/common/http/PaginationParams.kt [hunk 5]
```diff
@@ -20,17 +27,37 @@ open class PaginationParams @JvmOverloads constructor(
...
-    fun builder(): Builder<PaginationParams> {
-      return Builder(PaginationParams())
+    fun builder(): PaginationParamsBuilder<PaginationParams> {
+      return PaginationParamsBuilder(PaginationParams())
     }
   }
 
-  open class Builder<T : PaginationParams>(protected val params: T) {
+  /**
+   * Builder class for creating [PaginationParams]
```
Comment (mthadley) [inline workos/src/main/kotlin/com/workos/common/http/PaginationParams.kt:41]: Missing a period, I think.
```suggestion
   * Builder class for creating [PaginationParams].
```

File: workos/src/main/kotlin/com/workos/common/http/PaginationParams.kt [hunk 6]
```diff
@@ -20,17 +27,37 @@ open class PaginationParams @JvmOverloads constructor(
...
+   */
+  open class PaginationParamsBuilder<T : PaginationParams>(protected val params: T) {
+    /**
+     * Set the `after` query parameter.
+     */
     fun after(after: String) = apply { this.params["after"] = after }
+    /**
+     * Sets the `before` query parameter.
```
Comment (mthadley) [inline workos/src/main/kotlin/com/workos/common/http/PaginationParams.kt:49]: Noticed that the tense is inconsistent here: "Sets the..." vs "Set the...".

File: workos/src/main/kotlin/com/workos/common/http/RequestConfig.kt [hunk 7]
```diff
@@ -1,30 +1,55 @@
...
-      return Builder()
+    fun builder(): RequestConfigBuilder {
+      return RequestConfigBuilder()
     }
   }
 
-  class Builder {
-    var params: Map<String, String> = emptyMap()
+  /**
+   * Builder class for creating [RequestConfig]
```
Comment (mthadley) [inline workos/src/main/kotlin/com/workos/common/http/RequestConfig.kt:26]:
```suggestion
   * Builder class for creating [RequestConfig].
```

**Review by mthadley — COMMENTED** [review]
Just some small typos I noticed.

**Review by mthadley — APPROVED** [review]

**Pull request merged**